spectator/main/main.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+	"github.com/Netflix/spectator-go/spectator"
+	"github.com/Netflix/spectator-go/spectator/meter"
+	"strconv"
+	"time"
+)
+
+type Server struct {
+	registry       spectator.Registry
+	requestCountId *meter.Id
+	requestLatency *meter.Timer
+	responseSizes  *meter.DistributionSummary
+}
+
+type Request struct {
+	country string
+}
+
+type Response struct {
+	status int
+	size   int64
+}
+
+func (s *Server) Handle(request *Request) (res *Response) {
+	start := time.Now()
+
+	// initialize response
+	res = &Response{200, 64}
+
+	// Update the counter with dimensions based on the request.
+	tags := map[string]string{
+		"country": request.country,
+		"status":  strconv.Itoa(res.status),
+	}
+	requestCounterWithTags := s.requestCountId.WithTags(tags)
+	counter := s.registry.CounterWithId(requestCounterWithTags)
+	counter.Increment()
+
+	// ...
+	s.requestLatency.Record(time.Since(start))
+	s.responseSizes.Record(res.size)
+	return
+}
+
+func newServer(registry spectator.Registry) *Server {
+	return &Server{
+		registry,
+		registry.NewId("server.requestCount", nil),
+		registry.Timer("server.requestLatency", nil),
+		registry.DistributionSummary("server.responseSizes", nil),
+	}
+}
+
+func getNextRequest() *Request {
+	// ...
+	return &Request{"US"}
+}
+
+func main() {
```

Comment:
What's the thinking behind adding this code example to the repo?

Reply:
Oops forgot to delete